packages/safe-ds-vscode/src/extension/pythonServer.ts

```diff
@@ -60,20 +60,37 @@ export const startPythonServer = async function (): Promise<void> {
     logOutput('Started python server successfully');
 };
 
+/**
+ * Stop the python server process, if any currently exist. This will first try a graceful shutdown.
+ * If that fails, the whole process tree (starting at the child process spawned by startPythonServer) will get killed.
+ */
 export const stopPythonServer = async function (): Promise<void> {
+    logOutput('Stopping python server...');
     if (pythonServer !== undefined) {
         if (
             (pythonServerAcceptsConnections && !(await requestGracefulShutdown(2500))) ||
             !pythonServerAcceptsConnections
         ) {
-            treeKill(pythonServer.pid!);
+            logOutput(`Tree-killing python server process ${pythonServer.pid}...`);
+            const pid = pythonServer.pid!;
+            // Wait for tree-kill to finish killing the tree
+            await new Promise<void>((resolve, _reject) => {
+                treeKill(pid, (error) => {
+                    resolve();
+                    if (error !== undefined && error !== null) {
```

Comment:
```suggestion
                    if (error) {
```

It's not exactly the same, since it would not trigger if `error` is `false` or `""`, for example. But this is the usual way to check whether an object is defined.